python/tchannel/tornado/peer.py

```diff
@@ -458,35 +489,110 @@ def send(self, arg1, arg2, arg3, traceflag=False, headers=None):
             parent_span_id = None
             trace_id = None
 
-        # set 'as' to 'raw if 'as' is empty
+        # set default transport headers
         headers = headers or {}
-        if 'as' not in headers:
-            headers['as'] = 'raw'
-
-        connection = yield self.peer.connect()
-        message_id = connection.next_message_id()
-        response = yield connection.send_request(
-            Request(
-                service=self.service,
-                argstreams=[InMemStream(endpoint), arg2, arg3],
-                id=message_id,
-                headers=headers,
-                tracing=Trace(
-                    name=endpoint,
-                    trace_id=trace_id,
-                    parent_span_id=parent_span_id,
-                    endpoint=Endpoint(self.peer.host,
-                                      self.peer.port,
-                                      self.service),
-                    traceflags=traceflag,
-                )
+        real_header = {}
+        for h in transport_headers:
+            real_header[h] = headers.get(h, None) or self.headers.get(h, None)
```

Comment:
These are protocol header. I don't think it is a good idea to allow uses to set random things. Can TCP allow you set random header? not